### PR TITLE
fix(TimePickerCompat): letter key navigation should be case insensitive

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
## Previous Behavior
Press letter key in TimePicker input can navigate among item in listbox, but it is case sensitive.

## New Behavior
The navigation is now case insensitive.

## Related Issue(s)
#29824

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                     |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                Change |
| :-------------------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-timepicker-compat-preview</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `93.576 kB`<br />`31.394 kB` | `93.585 kB`<br />`31.404 kB` | `9 B` <br />`10 B`  |

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/6704fa058653887cb0f70dc78b1c593ee43b4594'>6704fa058653887cb0f70dc78b1c593ee43b4594</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by layershifter — APPROVED** [review]

File: packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx [hunk 1]
```diff
@@ -164,7 +164,7 @@ const useSelectTimeFromValue = (state: TimePickerState, callback: TimePickerProp
   React.useEffect(() => {
     if (freeform && value) {
       setActiveOption(prevActiveOption => {
-        if (prevActiveOption?.text?.indexOf(value) === 0) {
+        if (prevActiveOption?.text && prevActiveOption.text.toLowerCase().indexOf(value) === 0) {
```
Comment (ling1726) [inline packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx:167]: should both value and option text be compared using lower case?

Comment (YuanboXue-Amber) [reply packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx:167]: Oops _(:_」∠)_ let me update this and add a test

**Review by ling1726 — APPROVED** [review]

**Pull request merged**